### PR TITLE
Add dietary restrictions to registration form

### LINKED ISSUE #334: Add Dietary Restrictions
image

- Vegan, Vegetarian, Nut Allergy + any other common dietary restrictions

### PR DESCRIPTION
**Pull request by suhanic44**
Screen Shot 2023-09-01 at 7 32 27 PM

### CODE REVIEW

File: src/components/Forms/Register.jsx [hunk 1]
```diff
@@ -140,13 +147,23 @@ const Register = () => {
...
+              className="flex"
               text="Shirt Size"
               options={Shirts}
               field="shirt"
               user={user}
               setUser={setUser}
             />
           </Col>
+          <Col>
+            <Radio
```
Comment (shahdivyank) [inline src/components/Forms/Register.jsx]: change to checkbox

**Comment by shahdivyank** [comment]
assign urself

**Review by rfairooz — COMMENTED** [review]
Make the question Checkbox. Go ahead and remove the "other" option. Add in "Halal" as an option too

File: src/components/Radio.jsx [hunk 2]
```diff
@@ -10,10 +10,10 @@ const Radio = ({ text, field, options, user, setUser, editable = true }) => {
       <p className="mb-0">{text}</p>
       {!editable && <div>{user[field]}</div>}
       {editable && (
-        <div className="flex flex-wrap whitespace-nowrap">
+        <div className="flex flex-wrap whitespace-nowrap justify-between">
```
Comment (shahdivyank) [inline src/components/Radio.jsx]: remove edits for radio if they aren't being used

File: src/components/Forms/Register.jsx [hunk 3]
```diff
@@ -26,6 +26,44 @@ const Register = () => {
     },
   });
 
+  const [dietRestrictions, setDietRestrictions] = useState({
```
Comment (rfairooz) [inline src/components/Forms/Register.jsx]: Can you put this data in the Data/Register.js file and then import. Check out the Admin.jsx and AdminRequest.js on that. And also go ahead and update the screenshot of your work

**Review by rfairooz — APPROVED** [review]

File: src/components/Forms/Register.jsx [hunk 4]
```diff
@@ -14,6 +21,7 @@ import Button from "./Button";
 
 const Register = () => {
   const [user, setUser] = useState(User);
+  const [dietaryRestrictions, setDietRestrictions] = useState(dietRestrictions);
```
Comment (shahdivyank) [inline src/components/Forms/Register.jsx]:
```suggestion
  const [dietaryRestrictions, setDietRestrictions] = useState(DietRestrictions);
```

File: src/components/Forms/Register.jsx [hunk 5]
```diff
@@ -2,7 +2,14 @@
...
 import { useState } from "react";
 import User from "../../data/User";
-import { Ages, Majors, Grades, Genders, Shirts } from "../../data/Register";
+import {
+  Ages,
+  Majors,
+  Grades,
+  Genders,
+  Shirts,
+  dietRestrictions,
```
Comment (shahdivyank) [inline src/components/Forms/Register.jsx]:
```suggestion
  DietRestrictions,
```

File: src/data/Register.js [hunk 6]
```diff
@@ -28,6 +28,37 @@ export const Majors = [
   "Other",
 ];
 
+export const dietRestrictions = {
```
Comment (shahdivyank) [inline src/data/Register.js]:
```suggestion
export const DietRestrictions = {
```

**Review by shahdivyank — APPROVED** [review]

File: src/components/Forms/Register.jsx [hunk 7]
```diff
@@ -140,13 +155,27 @@ const Register = () => {
           </Col>
           <Col xl={12}>
             <Radio
+              className="flex"
```
Comment (shahdivyank) [inline src/components/Forms/Register.jsx]:
```suggestion
```

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**